packages/app/blockissuer/blockfactory/referenceprovider.go

```diff
@@ -39,6 +39,9 @@ func (r *ReferenceProvider) References(payload payload.Payload, strongParents mo
 
 	excludedConflictIDs := utxo.NewTransactionIDs()
 
+	r.protocol.Engine().Ledger.MemPool().ConflictDAG().WeightsMutex.Lock()
```

Comment:
Can we solve this differently? Locking this from the outside is not such a good idea

Reply:
It's because conflict tracking is done across multiple different components, and putting the lock in the ConflictDAG was easiest for access because the components that use the lock can access the shared lock. It  will be soon refactored anyway.